### ISSUE TITLE
Errors in Markdown highlighting

### ISSUE THREAD

**Issue by halflings** [jupyter/notebook#1080]
For an unknown reason, one of my cells has weird Markdown highlighting (see attached screenshot). It seems like it might have something to do with the presence of "%" characeters?
Other cells are formatted correctly.

![skarmavbild 2016-02-11 kl 12 08 31](https://cloud.githubusercontent.com/assets/1626936/12974957/70bdb52e-d0b8-11e5-8425-aeb332c6c320.png)

**Comment by halflings** [comment]
I updated the post because it seems that the dollar triggers these errors. Even escaping it (by adding a back-slash before the dollar sign) doesn't help.

**Comment by takluyver** [comment]
When Markdown cells are rendered, dollar signs can be used to delimit latex-style formulae. However, I didn't think that affected the syntax highlighting when they're in edit mode. :confused:

**Comment by willingc** [comment]
I did a bit of diving into this. I don't think that the "$" is the issue but rather how CodeMirror (or possibly our code) behaves when it reaches a special character. There is likely a regex in the CodeMirror code that causes % to be treated as the start of a <span class="cm-comment"> instead of the more common use of <span class="cm-delimit cm-delimit-open"> for the "$".

If any front end folks have a good suggestion to fix this, that would be great.

FWIW, the syntax highlighting is wonky but the rendered cell after hitting SHIFT-ENTER looks fine.

![screen shot 2016-02-11 at 4 48 51 pm](https://cloud.githubusercontent.com/assets/2680980/12996371/02290146-d0e4-11e5-9c33-9e3ea43bed4f.png)

![screen shot 2016-02-11 at 4 48 34 pm](https://cloud.githubusercontent.com/assets/2680980/12996370/01bcd5ac-d0e4-11e5-93ed-c172b5028e48.png)

**Comment by halflings** [comment]
The formatting looks OK (while editing) once I remove the dollar sign from
the example I've shown.

But yes, this is a code highlighting problem (that makes it hard to edit),
but the Markdown is still correctly rendered

fre 12 feb. 2016 02:29 Carol Willing <notifications@github.com> skrev:

**Comment by takluyver** [comment]
Someone also pointed out to me yesterday that if you put something that looks like a git commit ID in a markdown cell, it gets highlighted in blue and underlined. I guess the syntax highlighting is a bit more Github flavoured than the rendering.

**Comment by liviu-** [comment]
I also ran into a perhaps similar issue: 

![syntax_highlight-issue](http://i.imgur.com/B5b7AdJ.png)

Hope this helps

**Comment by jaypeedevlin** [comment]
This is something I run into almost daily, and while obviously the notebook renders fine, I do get annoyed at the minor syntax highlighting error.

Would this be something that would need to be fixed upstream (ie CodeMirror), or would this be handled inside Jupyter?

**Comment by magland** [comment]
This still appears to be an issue.

**Comment by tannert** [comment]
Looks like it works if you escape it with two backslashes.  I just tried it in Jupyter Lab 0.34.3 and it worked.

source: https://stackoverflow.com/questions/16089089/escaping-dollar-sign-in-ipython-notebook

**Comment by jasongrout** [comment]
In the latest jlab (0.35.4), I only have to escape dollar signs with one backslash, same as the classic notebook.

**Comment by jasongrout** [comment]
Ah, I didn't understand that the issue here is about the syntax highlighting, not the rendering. Yes, I still see an issue with the syntax highlighting.

**Comment by agnedil** [comment]
Two backslashes for $ and one for % did it. Looks fine for me in notebook server 5.7.4

**Comment by austincauthon** [comment]
A workaround for this issue is to use the unicode `&#0036;` in place of using a `$`. This will render a `$` inside your rendered document, but not affect your syntax highlighting in your editor.

**Comment by mobilinkd** [comment]
Using &#0036; does not fix the problem in the version 5.7.8.  Worse, there does not seem to be an escaping mechanism that works across Jupyter, Github's notebook renderer, and nbviewer.  The fact that jupyter and nbviewer disagree about markup rendering is a bit frustrating.  The notebook renderer requires double-backslashes to escape $ but nbviewer will show a \$ for some of these.

**Comment by dcompoze** [comment]
The `$` sign breaks the markdown cell syntax highlighting (in edit mode) and the escape sequences don't seem to make a difference.

Also, it's not possible to escape it when you write it in a code block:

![selection-1644195339](https://user-images.githubusercontent.com/76269710/152709591-cfa724af-0df4-4088-b4c9-1113336efb48.png)

![selection-1644195339](https://user-images.githubusercontent.com/76269710/152709580-3b57f5ec-18cc-4f50-b287-2b2d554a7d9c.png)

Though the symbol renders correctly, which suggests that the highlighting parser has different logic than the renderer.

**Comment by shepware** [comment]
I'm finding that if I do "\\$" rather than just typing "$" in the context of writing a paragraph, it makes things look good (within Jupyter Notebook, Python 3, using narrative, Esc+M to set Markdown type cells that present well if you hit Enter after typing.  Unfortunately, after you "git add .", "git commit -m 'comment'", and "git push origin master", the rendering at github.com ends up doing some odd font intermittently between instances of "\\$" in my text narrative, failing to ignore the stylistic implications of "$" in the way that jupyter ignored in the version I last reviewed before pushing to git.  

TLDR: markdown in jupyter looks like hell on github

**Comment by tvbush** [comment]
Well, this really doesn't work. I don't see any solution at all. I've tried everything suggested on this page. The only one that works is where there is only a single dollar sign on a line. Anything else seems to break, sometimes in VERY strange ways. Note that when this is rendered, both Chrome and Firefox display the second line correctly for a moment, and then it is turned into math-renderer style which is clearly NOT what I want.

My first image shows the source in plain text. The file is named "EXAMPLE.md".

![Screen Shot 2022-06-09 at 3 51 51 PM](https://user-images.githubusercontent.com/99690003/172958781-62a0a6fc-1ab8-47d6-90c1-4e3d1e2a5490.png)

My second image shows how it is rendered on Github.

![Screen Shot 2022-06-09 at 3 51 09 PM](https://user-images.githubusercontent.com/99690003/172958798-bbbf92d8-5032-418c-96b9-6dc98fd3e0bb.png)

**Comment by jasongrout** [comment]
> how it is rendered on Github.

@tvbush - we don't have any control over how GitHub renders Jupyter notebooks. If this is an issue with it displaying a notebook on GitHub, you'll need to file a support request with GitHub.

**Comment by tvbush** [comment]
Did you try to reproduce my issue?
Yes, I will send it to Github too. Thanks!
I was just happy to find someone else who had similar issues!

**Comment by jasongrout** [comment]
This renders correctly in the latest JupyterLab and Jupyter Notebook:

```
A test \\${asdf} \\${asdflkj}
```
<img width="179" alt="Screen Shot 2022-06-10 at 18 12 44" src="https://user-images.githubusercontent.com/192614/173164668-d754afea-9324-470d-8800-c9722f45c91f.png">
